03-viz.Rmd

```diff
@@ -346,7 +489,10 @@ morley_hist <- ggplot(morley, aes(x = Speed, fill = factor(Expt))) +
 morley_hist
 ```
 
-Unfortunately, the attempt to separate out the experiment number visually has created a bit of a mess. All of the colours
+
+> Notice that we wrote `factor(Expt)` in the' fill' aesthetic mapping. A **factor** helps us represent categorical data. Factors can be ordered or unordered, and even though factors look like characters, they are integers underneath. They contain a set of values, known as **levels**. Storing something as a factor helps ensure that R will treat the data correctly. For instance, if we look at the printed `morley` data, the column `Expt` is an integer (we see the label `<int>` underneath the `Expt` column name). But, we want to treat it as a categorical variable, so we use the function `factor()` to make sure it is treated as such. 
```

Comment:
I think factors should be explained/introduced in the wrangling chapter with the other vector types, and then rediscussed here. In particular, they impact plots two ways - ensuring a colour (or other channel) is mapped discretely instead of qualitatively when appropriate, as well as the ordering of things in a plot. `reorder` and `arrange` have quite different influences on a plot (with the former having a lot, and the latter having almost none).